### PR TITLE
Fix pod's network label priority incorrect

### PR DESCRIPTION
**Pull request by kebe7jun**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/42675

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go [hunk 1]
```diff
@@ -76,6 +76,8 @@ func NewEndpointBuilder(c controllerInterface, pod *v1.Pod) *EndpointBuilder {
 		namespace:    namespace,
 		hostname:     hostname,
 		subDomain:    subdomain,
+		// todo(kebe7jun) use ISTIO_META_NETWORK instead?
+		metaNetwork: network.ID(podLabels[label.TopologyNetwork.Name]),
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go:80]: This is not correct, the original intention isto prefer the network from metadata which is from `ISTIO_META_NETWORK`. 

But it is changing to respect pod label first

Comment (kebe7jun) [reply pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go]: The implementation was changed.

Comment (stevenctl) [reply pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go]: On the pods with this bug, what does there ISTIO_META value hold?

If it's different from the label that will cause issues.

If it's empty, we should only fallback to the label.

File: releasenotes/notes/fix-42675.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# issue is a list of GitHub issues resolved in this note.
+# If issue is not in the current repo, specify its full URL instead.
+issue:
+  - 42675
+
+# releaseNotes is a markdown listing of any user facing changes. This will appear in the
+# release notes.
+releaseNotes:
+  - |-
+    **Fixed** Namespace's network label has higher priority than the Pod's network label.
```
Comment (stevenctl) [inline releasenotes/notes/fix-42675.yaml:34]:
```suggestion
    **Fixed** a bug that caused the Namespace's network label to have a higher priority than the Pod's network label.
```

The original phrasing was ambiguous to whether this is the new or old behavior.

**Review by stevenctl — APPROVED** [review]

**Pull request merged**